compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp

```diff
@@ -43,17 +43,16 @@ namespace cppmicroservices
             std::shared_ptr<cppmicroservices::async::AsyncWorkService> asyncWorkSvc,
             std::shared_ptr<SCRExtensionRegistry> extensionReg)
             : tokenCounter(0)
-            , bundleContext(context)
-            , logger(std::move(logger))
-            , asyncWorkService(asyncWorkSvc)
-            , extensionRegistry(extensionReg)
-        {
-            if (!bundleContext || !(this->logger) || (!this->asyncWorkService) || (!this->extensionRegistry))
+            , logger(logger)
+            , componentFactory(std::make_shared<ComponentFactoryImpl>(context, logger, asyncWorkSvc, extensionReg))
+ 
+          {
+            if (!context || !(this->logger) || !asyncWorkSvc || !extensionReg || !componentFactory)
```

Comment:
This check is already done in ComponentFactoryImpl::ComponentFactoryImpl. It's redundant here.

Reply:
We've been following the "standard" in DS that each constructor checks its own input parameters.

Reply:
OK. I'd argue that in this case since those values are not stored in the ConfigurationNotifier but are just passed on to the ComponentFactoryImpl constructor which is what stores them and uses them that that class should be the one to decide what valid arguments are for itself.